clients/src/test/java/org/apache/kafka/clients/consumer/internals/MembershipManagerImplTest.java

```diff
@@ -141,23 +136,41 @@ private MembershipManagerImpl createMembershipManagerJoiningGroup(String groupIn
     }
 
     private MembershipManagerImpl createMembershipManager(String groupInstanceId) {
-        return spy(new MembershipManagerImpl(
+        return new MembershipManagerImpl(
             GROUP_ID, Optional.ofNullable(groupInstanceId), REBALANCE_TIMEOUT, Optional.empty(),
             subscriptionState, commitRequestManager, metadata, logContext, Optional.empty(),
-            backgroundEventHandler, time, rebalanceMetricsManager));
+            backgroundEventHandler, time, rebalanceMetricsManager);
     }
 
     private MembershipManagerImpl createMembershipManagerJoiningGroup(String groupInstanceId,
                                                                       String serverAssignor) {
-        MembershipManagerImpl manager = spy(new MembershipManagerImpl(
+        MembershipManagerImpl manager = new MembershipManagerImpl(
                 GROUP_ID, Optional.ofNullable(groupInstanceId), REBALANCE_TIMEOUT,
                 Optional.ofNullable(serverAssignor), subscriptionState, commitRequestManager,
                 metadata, logContext, Optional.empty(), backgroundEventHandler, time,
-                rebalanceMetricsManager));
+                rebalanceMetricsManager);
         manager.transitionToJoining();
         return manager;
     }
 
+    private void createCommitRequestManager(boolean autoCommit) {
+        ConsumerConfig config = mock(ConsumerConfig.class);
+        if (autoCommit) {
+            when(config.getBoolean(ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG)).thenReturn(true);
+        }
+
+        commitRequestManager = new CommitRequestManager(
```

Comment:
I would aim for not having an actual commitRequestManager here, a mock should do, but I may be missing something. Check my other comments to get rid of the `createCommitRequestManager` (and maybe we can end up removing this?)